plugins/TPChannelFilter.cpp

```diff
@@ -0,0 +1,126 @@
+/**
+ * @file TPChannelFilter.cpp
+ *
+ * This is part of the DUNE DAQ Application Framework, copyright 2021.
+ * Licensing/copyright details are in the COPYING file that you should have
+ * received with this code.
+ */
+
+#include "TPChannelFilter.hpp"
+#include "triggeralgs/TriggerPrimitive.hpp"
+
+#include <string>
+
+namespace dunedaq {
+namespace trigger {
+TPChannelFilter::TPChannelFilter(const std::string& name)
+  : DAQModule(name)
+  , m_thread(std::bind(&TPChannelFilter::do_work, this, std::placeholders::_1))
+  , m_input_queue(nullptr)
+  , m_output_queue(nullptr)
+  , m_queue_timeout(100)
+{
+
+  register_command("conf", &TPChannelFilter::do_conf);
+  register_command("start", &TPChannelFilter::do_start);
+  register_command("stop", &TPChannelFilter::do_stop);
+  register_command("scrap", &TPChannelFilter::do_scrap);
+}
+
+void
+TPChannelFilter::init(const nlohmann::json& iniobj)
+{
+  try {
+    m_input_queue.reset(new source_t(appfwk::queue_inst(iniobj, "tpset_source")));
+    m_output_queue.reset(new sink_t(appfwk::queue_inst(iniobj, "tpset_sink")));
+  } catch (const ers::Issue& excpt) {
+    throw dunedaq::trigger::InvalidQueueFatalError(ERS_HERE, get_name(), "input/output", excpt);
+  }
+}
+
+void
+TPChannelFilter::get_info(opmonlib::InfoCollector& /* ci */, int /*level*/)
+{}
+
+void
+TPChannelFilter::do_conf(const nlohmann::json& conf_arg)
+{
+  m_conf = conf_arg.get<dunedaq::trigger::tpchannelfilter::Conf>();
+  m_channel_map = dunedaq::detchannelmaps::make_map(m_conf.channel_map_name);
+}
+
+void
+TPChannelFilter::do_start(const nlohmann::json&)
+{
+  m_thread.start_working_thread("channelfilter");
+  TLOG_DEBUG(2) << get_name() + " successfully started.";
+}
+
+void
+TPChannelFilter::do_stop(const nlohmann::json&)
+{
+  m_thread.stop_working_thread();
+  TLOG_DEBUG(2) << get_name() + " successfully stopped.";
+}
+
+void
+TPChannelFilter::do_scrap(const nlohmann::json&)
+{}
+
+bool
+TPChannelFilter::channel_should_be_removed(int channel) const
+{
+  // The plane numbering convention is found in detchannelmaps/plugins/VDColdboxChannelMap.cpp and is:
+  // U = 0, Y = 1, Z = 2
+  uint plane = m_channel_map->get_plane_from_offline_channel(channel);
+  return ((plane == 0 || plane == 1) && m_conf.keep_induction) || (plane == 2 && m_conf.keep_collection);
```

Comment:
I'm confused by this logic. In my testing, I ran with both keep_induction and keep_collection false, but no TPs were removed. My supposition is that if the plane is not 0, 1, or 2, then this will return false and the channel will not be removed. The code might be more clearly expressed as follows:
```C++
// U = 0, Y = 1, Z =2
// Check for collection
if(plane == 0 || plane == 1) return !m_conf.keep_collection
// Check for induction
if(plane == 2) return !m_conf.keep_induction;
// Unknown plane?!
TLOG() << "Encountered unexpected plane " << plane << " from channel " << channel << ", check channel map?";
return false;
```

Reply:
You're right - I had a sign error. After a couple of attempts I think I've finally got it right. I also added handling for plane number 9999, which indicates channels in the VD coldbox that aren't actually connected to a wire - we unconditionally drop those.